GIthub Login

berry-13 opened the pull request:
﻿Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

I have added the ability to log in with GitHub. I didn't understand, even by looking at the OpenID PR, why it displays the double "or" when both Google and GitHub are enabled. Another thing is that authentication occurs through email, so if someone logs in with GitHub first and then tries to reset the password, they will be able to access the account both through GitHub and with email and password, even if they didn't register through the sign-up button. 
Actually, this functionality is similar to the one of OpenAI. In fact, if you log in through Google, you won't be able to register your account anymore using the sign-up button. 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
##

I tried to run it with docker desktop on windows and on linux with docker-compose

### **Test Configuration**:
##


## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules


![image](https://github.com/danny-avila/LibreChat/assets/81851188/19cb5d63-40a1-439b-8879-43fee736a5b6)
![image](https://github.com/danny-avila/LibreChat/assets/81851188/da877f06-c301-4eb9-a3a1-78b6cc94e5ec)
![image](https://github.com/danny-avila/LibreChat/assets/81851188/eef01e1e-cc52-4c53-8fea-b7353ab8ff77)

berry-13 committed:
Add files via upload

berry-13 committed:
Create linode-setup.md

berry-13 committed:
Create cloudflare-setup.md

berry-13 committed:
Update cloudflare-setup.md

berry-13 committed:
Delete 4-linode.png

berry-13 committed:
Delete 3-linode.png

berry-13 committed:
Add files via upload

berry-13 committed:
Add files via upload

berry-13 committed:
Update cloudflare-setup.md

berry-13 committed:
Update linode-setup.md

berry-13 committed:
Rename cloudflare-setup.md to cloudflare.md

berry-13 committed:
Rename linode-setup.md to linode.md

berry-13 committed:
Update mkdocs.yml

berry-13 committed:
Update cloudflare.md

berry-13 committed:
Update linode.md

berry-13 committed:
Update README.md

berry-13 committed:
Update README.md

berry-13 committed:
Update linode.md

sentence in Italian

berry-13 committed:
Merge branch 'danny-avila:main' into main

berry-13 committed:
Merge branch 'danny-avila:main' into main

berry-13 committed:
v1

The frontend has been completed, along with the .env variables.

However, there is an issue of infinite loading thereafter.

berry-13 committed:
Fix email and remove deprecated GitHub passport

berry-13 committed:
Merge branch 'danny-avila:main' into main

berry-13 committed:
Update user_auth_system.md

add How to Set Up a Github Authentication

berry-13 committed:
Update .env.example

Improved the comment above the GitHub client ID and secret.

berry-13 committed:
Update user_auth_system.md

berry-13 committed:
Update package.json

berry-13 committed:
Remove unnecessary passport GitHub package

berry-13 committed:
Merge branch 'main' of http://github.com/Berry-13/librechat

berry-13 committed:
Merge branch 'danny-avila:main' into main

danny-avila commented:
Appreciate this addition, will definitely use it! Going to test this soon.

fuegovic commented:
@danny-avila 

---

I tested the GitHub login feature this morning before the PR was made, and it works great! I had no issues at all, and the instructions to set it up are clear.

### Note:
The only thing to be aware of is that GitHub uses an email profile when creating a user.

## Testing:
- I tested on a DB with an existing profile using the same email as GitHub, and everything was flawless. My profile was intact and I had another option to log in to that profile.

- Then I tested on a new DB, creating a GitHub user first, then trying to create a user with the same email as GitHub. I got an error when trying to create the email profile, **which was expected** since the user already existed. Then I used the password reset option to set up a new password for the email login (since it was not set by GitHub) and once again everything was flawless. I was able to log in to that profile using two different methods of authentication.

## Impressions:
I personally like that both login options gave me access to the same email profile. It was a very painless experience.

berry-13 committed:
fixed conflicts

 fixed conflicts between Berry-13:main and danny-avila:main

in api/server/index.js 45:54

berry-13 committed:
Delete e -i HEAD~2

berry-13 committed:
Merge branch 'main' into main
